parquet/src/main/java/org/apache/iceberg/parquet/ParquetReader.java

```diff
@@ -130,13 +139,23 @@ private void advance() {
 
       PageReadStore pages;
       try {
-        pages = reader.readNextRowGroup();
+        // Because of the issue of PARQUET-1901, we cannot blindly call readNextFilteredRowGroup()
+        if (hasRecordFilter) {
+          pages = reader.readNextFilteredRowGroup();
```

Comment:
Iceberg is evaluating filters for row groups itself and skipping them (just a few lines before this), how does this change improve upon what Iceberg already does? (I am not too familiar with Parquet, so I may be totally missing the point)

Reply:
I read more into the code and I think I understand the benefit here. So a Parquet row group is made up of multiple pages, previously we were only skipping unmatched row groups, now we can skip unmatched row groups and unmatched pages. Let me know if this seems correct.